File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int low = 0;
        int medium = 0;
        int high = nums.length-1;
        while(medium<=high)
        {
            if(nums[medium]==0)
            {
                int temp = nums[medium];
                nums[medium] = nums[low];
                nums[low] = temp;
                medium++;
                low++;
            }
            else if(nums[medium]==1)
            {
                medium++;
            }
            else if(nums[medium]==2)
            {
                int temp = nums[medium];
                nums[medium]= nums[high];
                nums[high] = temp;
                high--;
            }
        }
    }
}

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int start = 0;
        int mid = 0;
        int end = nums.size()-1;

        while(mid<=end){
            if(nums[mid] == 0){
                int temp = nums[mid];
                nums[mid] = nums[start];
                nums[start] = temp;
                mid++;
                start++;
            }
            else if(nums[mid] == 1){
                mid++;
            }
            else if(nums[mid] == 2){
                int temp = nums[mid];
                nums[mid] = nums[end];
                nums[end] = temp;
                end--;
            }
        }
    }
};